src/Symfony/Component/HttpClient/ServerSentEvents/EventSource.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Symfony\Component\HttpClient\ServerSentEvents;
+
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\HttpClient\HttpClientTrait;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Symfony\Contracts\HttpClient\ResponseStreamInterface;
+
+final class EventSource implements HttpClientInterface
```

Comment:
Hello everyone. I have some feedback here, not about the nature of this work but the naming.

This naming feels so weird to me, let me explain why:
1. The namespace is named ServerSentEvents while this looks related to Mercure only (I know SSE is a standard, but do you have another tool that will work with? Although Mercure adds its own layer that is not standard)
2. Naming this class `EventSource` while it's not really related to the event sourcing approach feels so weird. I get that you may use it to source events, but you can't consider it as a good practice for the most global _EventSourcing_ approach (due to the lack of persistence of events)

I'm sorry I'm posting these ideas without a better naming suggestion. I think this is at least good to highlight things and trigger some reflection.

Reply:
It's for the SSE nothing related to mercure here there are many other applications to server sent events (eg websocket replacement).  

> Naming this class EventSource while it's not really related to the event sourcing approach feels so weird.

this is just because it was meant to be an implementation of the [EventSource interface](https://www.w3.org/TR/2009/WD-eventsource-20090421/#the-eventsource-interface) and it's called `Eventsource` in javascript, python, rust, golang etc..

Reply:
Well, this get rid about the naming question.

_(SSE are great; but they are not WebSocket replacement)_

Btw may be a stupid question but why not linking [this RFC](https://www.w3.org/TR/eventsource/#eventsource) which is not a draft?

Reply:
thanks I changed the link, I don't see where you couldn't replace websockets by SSE but that's another topic.

Reply:
websocket is a bidirectional channel. SSE is unidirectional.

A replacement of websocket might be SSE+API, but not SSE only.